benchmarking/benchmarking.md

```diff
@@ -0,0 +1,103 @@
+# Benchmarking Harness
+
+Author: @biswapanda
+
+# Problem statement:
+In current state, benchmarking has a few problems:
+
+1. UX: Experimentation, debugging, and iteration is hard. 
+use case: As a user, I want to easily experiment with different configs, and get results quickly and compare them.
+
+2. Reproducibility is hard: we don't store the input configs and results.
+use case: As a user, I want to be able to reproduce my experiments and share them with others.
+
+3. benchmarking steps are tightly coupled. If a sinlge step/benchmark config fails the entire process is aborted/retried.   
+
+4. port-forwarding and benchmarking has non-deterministic latency characteristics. 
+
+## Proposed plan:
+
+1. decouple all steps and then compose them together: prep a model, deploy k8s cr, benchmark, collect data 
+
+2. capture configs for the experiment: deploy (config or a reference to deployment), benchmark, model etc
+
+3. we'd run benchmarks inside k8s cluster in k8s native approach.
+
+
+## Steps:
+Following steps are executed by the harness:
+
+Note: These steps are reusable across different tests (LLM benchmarking, Accuracy testing, Functional testing etc)
+
+Since these steps are reusable across different tests, we can swap the container used for each step.
+
+1. Initialize experiment
+
+    a. (Optional) deploy model
+
+    b. wait for the model to be ready
+
+2. Run Benchmarking test using configs and benchmark container (genai-perf, ai perf or 3rd party tool)
```

Comment:
I think this is hand-waving a lot of other configurations which need to happen for the particular setup. If the user wants aggregated vs disaggregated, if disaggregated, then how many workers in each? What is the gpu allocation for each of the workers, etc.?

How does this proposal address this, or are you making assumptions about the environment or use case? If so, that's fine, and I would like that explanation here.